superset/assets/javascripts/explorev2/components/ExploreViewContainer.jsx

```diff
@@ -25,9 +26,20 @@ class ExploreViewContainer extends React.Component {
     };
   }
 
-  onQuery() {
+  componentWillReceiveProps(nextProps) {
+    let refreshChart = false;
+    autoQueryFields.forEach((field) => {
+      if (nextProps.form_data[field] !== this.props.form_data[field]) {
+        refreshChart = true;
```

Comment:
should we break this for loop once we set this as true? although we only have a couple fields in the `autoQueryFields` array so it won't really make much of a difference.